solidity/ecdsa/test/WalletRegistry.RandomBeacon.test.ts

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-underscore-dangle */
+import { ethers, waffle, helpers } from "hardhat"
+import { expect } from "chai"
+
+import { dkgState, walletRegistryFixture } from "./fixtures"
+import { fakeRandomBeacon, resetMock } from "./utils/randomBeacon"
+
+import type { FakeContract } from "@defi-wonderland/smock"
+import type { ContractTransaction } from "ethers"
+import type {
+  WalletRegistry,
+  WalletRegistryStub,
+  IRandomBeacon,
+} from "../typechain"
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+
+const { createSnapshot, restoreSnapshot } = helpers.snapshot
+
+describe("WalletRegistry - Random Beacon", async () => {
+  let walletRegistry: WalletRegistryStub & WalletRegistry
+  let randomBeacon: FakeContract<IRandomBeacon>
+
+  let walletOwner: SignerWithAddress
+  let thirdParty: SignerWithAddress
+
+  before("load test fixture", async () => {
+    await createSnapshot()
+
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({ walletRegistry, walletOwner, thirdParty } = await waffle.loadFixture(
+      walletRegistryFixture
+    ))
+
+    randomBeacon = await fakeRandomBeacon(walletRegistry)
+  })
+
+  after(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("requestNewWallet", async () => {
+    context("when requestRelayEntry reverts", async () => {
+      let tx: Promise<ContractTransaction>
+
+      before(async () => {
+        await createSnapshot()
+
+        await randomBeacon.requestRelayEntry.reverts("beacon internal error")
+
+        tx = walletRegistry.connect(walletOwner).requestNewWallet()
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+
+        resetMock(randomBeacon)
+      })
+
+      it("should revert", async () => {
+        // FIXME: For some reason this check doesn't work with the expected error message
```

Comment:
Not very familiar with a smock lib (but I like the mocking lib idea) but can it be related to a wrong param, ie shouldn't we use `requestRelayEntry.whenCalledWith(giveWalletRegistryAddress)` and the reason for not failing is becuase of the addresses in the params mismatch?